Geometry_Engine/Modify/RemoveShortSegments.cs

```diff
@@ -67,17 +63,27 @@ public static Polyline RemoveShortSegments(this Polyline polyline, double minimu
                     startIndex++; //Move onto the next point
             }
 
-            if (pnts.First() != pnts.Last())
-                pnts.Add(pnts.First()); //Reclose polyline
+            if (!polyline.IsClosed())
+            {
+                if (pnts.Last() != originalPnts.Last())
+                {
+                    pnts.Remove(pnts.Last());
+                    pnts.Add(originalPnts.Last()); //Keep the original endpoint of the polyline                   
+                }
+            }
+
+            if (polyline.IsClosed())
```

Comment:
See previous comment on the other method about using an `else` instead of two `IsClosed` checks